plugins/network-elements/juniper-srx/src/com/cloud/network/resource/JuniperSrxResource.java

```diff
@@ -2841,8 +2841,9 @@ private boolean manageSecurityPolicy(SecurityPolicyType type, SrxCommand command
                             action = "<permit></permit>";
                         }
 
-                        xml = replaceXmlValue(xml, "action", action);
+                        
                     }
+                    xml = replaceXmlValue(xml, "action", action);
```

Comment:
Why did you put it out of "if-then-else"? 
Shouldn't it only be executed only if "!type.equals(SecurityPolicyType.SECURITYPOLICY_EGRESS_DEFAULT)"?